src/songMetadata/musicBrainzModule/musicBrainzClient.ts

```diff
@@ -0,0 +1,117 @@
+ * ----------------------------------------------
+ * Decided to make my own version, since it's not used widely
+ * For this specific use case this will do.
+ */
+
+import { SearchRecordingsResult } from './musicBrainzResponseTypes';
+
+export enum SearchReleasesFields {
+    artistId = 'arid', // artist id
+    artistNameOnRecording = 'artist', // artist name is name(s) as it appears on the recording
+    artistName = 'artistname', // an artist on the recording, each artist added as a separate field
+    creditName = 'creditname', // name credit on the recording, each artist added as a separate field
+    commentDisambiguation = 'comment', // recording disambiguation comment
+    country = 'country', // recording release country
+    date = 'date', // recording release date
+    durationMs = 'dur', // duration of track in milliseconds
+    format = 'format', // recording release format
+    isrc = 'isrc', // ISRC of recording
+    trackNumberFreeText = 'number', // free text track number
+    position = 'position', // the medium that the recording should be found on, first medium is position 1
+    primaryType = 'primarytype', // primary type of the release group (album, single, ep, other)
+    qdur = 'qdur', // quantized duration (duration / 2000)
+    recordingId = 'rid', // recording id
+    recordingName = 'recording', // name of recording or a track associated with the recording
+    recordingAccent = 'recordingaccent', // name of the recording with any accent characters retained
+    releaseId = 'reid', // release id
+    releaseName = 'release', // release name
+    releaseGroupId = 'rgid', // release group id
+    secondaryType = 'secondarytype', // secondary type of the release group (audiobook, compilation, interview, live, remix soundtrack, spokenword)
+    status = 'status', // Release status (official, promotion, Bootleg, Pseudo-Release)
+    trackId = 'tid', // track id
+    trackNumberOnMedium = 'tnum', // track number on medium
+    tracksCount = 'tracks', // number of tracks in the medium on release
+    tracksCountOnRelease = 'tracksrelease', // number of tracks on release as a whole
+    tag = 'tag', // folksonomy tag
+    type = 'type', // type of the release group, old type mapping for when we did not have separate primary and secondary types or use standalone for standalone recordings
+    video = 'video', // true to only show video tracks
+}
+
+export type SearchReleasesCriteria = {
+    field: SearchReleasesFields;
+    value: string;
+}[];
+
+class MusicBrainzUrlHelper {
+    public static getSearchReleasesQuery<T>(
+        searchFields: SearchReleasesCriteria,
+    ): string {
+        return `?query=${searchFields.map((f) => {
+            return `${f.field}:"${encodeURIComponent(f.value)}"`;
+        })}`;
+    }
+
+    public static getGeneralQueryString(
+        offset?: number,
+        limit?: number,
+    ): string {
+        let result = '';
+        if (offset !== undefined) {
```

Comment:
Why not just `if (offset)`?

Reply:
Because, you can pass 0 ("falsy" value). In my opinion this should be reflected in query parameters.